frontend/apps/reader/modules/readerdevicestatus.lua

```diff
@@ -27,24 +29,29 @@ function ReaderDeviceStatus:init()
                     powerd:setDismissBatteryStatus(false)
                 end
             else
+                if self.battery_confirm_box then
+                    UIManager:close(self.battery_confirm_box)
```

Comment:
May be you can set `self.battery_confirm_box =nil` after this - so if you don't display a new one, next time, you're not UIManager:close() something that was already (even if UIManager doesn't mind).
You could also do the same in the ok/cancel_callback to avoid having this step here done when the user have closed it - but if you don't, it will just be a single unuseful UIManager:close() instead of many, which I can live with :).

Reply:
I think of combining two battery level alerts into one.
They distinguish with a word only: Low and High. The code is duplicated.
Something like "Battery level alert: 17%".
A user can understand which threshold is exceeded, can't he?

Reply:
Can't you know in there / store the fact whether the last trigger is high or low ? So you could adapt the message ?